django/db/backends/ddl_references.py

```diff
@@ -83,10 +83,14 @@ def __init__(self, table, columns, quote_name, col_suffixes=()):
 
     def __str__(self):
         def col_str(column, idx):
+            col = self.quote_name(column)
             try:
-                return self.quote_name(column) + self.col_suffixes[idx]
+                suffix = self.col_suffixes[idx]
+                if suffix:
+                    col = '{} {}'.format(col, suffix)
```

Comment:
We added checking `suffix` so I don't see any value in catching `IndexError` anymore. We can simplify `col_str()` with `get()` in both cases, e.g.
```python
def col_str(column, idx):
    col = self.quote_name(column)
    suffix = self.col_suffixes.get(idx)
    if suffix:
        col = '{} {}'.format(col, suffix)
    return col
```
and
```python
def col_str(column, idx):
    # Index.__init__() guarantees that self.opclasses is the same
    # length as self.columns.
    col = '{} {}'.format(self.quote_name(column), self.opclasses[idx])
    suffix = self.col_suffixes.get(idx)
    if suffix:
        col = '{} {}'.format(col, suffix)
    return col
```